shared/test/test_poste.rb

```diff
@@ -1,57 +1,69 @@
+  STUDENT_EMAIL = 'student@example.net'.freeze
+  STUDENT_EMAIL_HASH = Digest::MD5.hexdigest(STUDENT_EMAIL).freeze
+  TEACHER_EMAIL = 'teacher@example.net'.freeze
+  TEACHER_EMAIL_HASH = Digest::MD5.hexdigest(TEACHER_EMAIL).freeze
+
   def test_unsubscribe_for_existing_contact
     email = 'existing@example.net'
+    hashed_email = Digest::MD5.hexdigest(email)
+
     Poste2.create_recipient(email, {ip_address: '5.6.7.8.'})
+    assert POSTE_DB[:contacts].where(hashed_email: hashed_email).first
+
     Poste.unsubscribe(email, Digest::MD5.hexdigest(email))
-    assert POSTE_DB[:contacts].where(email: email).first[:unsubscribed_at]
+    assert POSTE_DB[:contacts].
+      where(hashed_email: hashed_email).first[:unsubscribed_at]
   end
 
   def test_unsubscribe_for_new_contact
     email = 'new@example.net'
+    hashed_email = Digest::MD5.hexdigest(email)
+
+    assert POSTE_DB[:contacts].where(hashed_email: hashed_email).empty?
     Poste.unsubscribe(
       email, Digest::MD5.hexdigest(email), {ip_address: '5.6.7.8.'})
-    assert POSTE_DB[:contacts].where(email: email).first[:unsubscribed_at]
+    assert POSTE_DB[:contacts].
+      where(hashed_email: hashed_email).first[:unsubscribed_at]
   end
 
   def test_unsubscribe_for_code_studio_student_stores_no_email
-    email = 'student@example.net'
-    hashed_email = Digest::MD5.hexdigest(email)
-    DASHBOARD_DB[:users].insert(
-      email: email,
-      hashed_email: hashed_email,
-      username: 'code studio student',
-      user_type: 'student',
-      birthday: '2000-01-02'
-    )
-
-    Poste.unsubscribe(email, hashed_email, {ip_address: '1.2.3.4'})
+    Poste.unsubscribe(
+      STUDENT_EMAIL, STUDENT_EMAIL_HASH, {ip_address: '1.2.3.4'})
 
     assert POSTE_DB[:contacts].
-      where(hashed_email: hashed_email).first[:unsubscribed_at]
-    assert_nil POSTE_DB[:contacts].where(email: email).first
+      where(hashed_email: STUDENT_EMAIL_HASH).first[:unsubscribed_at]
+    assert_nil POSTE_DB[:contacts].where(email: STUDENT_EMAIL).first
   end
 
   def test_unsubscribe_for_code_studio_teacher_stores_email
-    email = 'teacher@example.net'
+    Poste.unsubscribe(
+      TEACHER_EMAIL, TEACHER_EMAIL_HASH, {ip_address: '1.2.3.4'})
+
+    assert POSTE_DB[:contacts].
+      where(hashed_email: TEACHER_EMAIL_HASH).first[:unsubscribed_at]
+    assert POSTE_DB[:contacts].
+      where(email: TEACHER_EMAIL).first[:unsubscribed_at]
+  end
+
+  def test_dashboard_student_for_nonuser
+    email = 'nonexistent_email@fake_domain.com'
     hashed_email = Digest::MD5.hexdigest(email)
-    DASHBOARD_DB[:users].insert(
-      email: email,
-      hashed_email: hashed_email,
-      username: 'code studio teacher',
-      user_type: 'teacher',
-      birthday: '1990-01-02'
-    )
+    assert !Poste.dashboard_student?(hashed_email)
+  end
 
-    Poste.unsubscribe(email, hashed_email, {ip_address: '1.2.3.4'})
+  def test_dashboard_student_for_student
+    assert Poste.dashboard_student?(STUDENT_EMAIL_HASH)
```

Comment:
same as above. Am I missing where the `STUDENT_EMAIL_HASH` is added to dashboard?